models/Asset/Image/Thumbnail/Processor.php

```diff
@@ -472,6 +475,27 @@ public static function process(Asset $asset, Config $config, $fileSystemPath = n
         ];
     }
 
+    /**
+     * @throws \Exception
+     */
+    private static function addOrUpdateThumbnailCache(
+        Asset $asset,
+        Config $config,
+        string $filename
+    ): void
+    {
+        $tmpFsPath = $asset->getLocalFile();
```

Comment:
This is not doing the same as it was before, because it always get's the dimensions from the original asset and not from the thumbnail ...

Reply:
fixed with new approach